pkg/resources/management.cattle.io/v3/globalrole/validator.go

```diff
@@ -121,51 +122,55 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
 		return nil, err
 	}
 
-	// check for escalation separately between cluster permissions and global permissions to prevent crossover
+	var returnError error
+
+	// Validate the global and namespaced rules of the new GR
+	globalRules := a.grbResolver.GlobalRoleResolver.GlobalRulesFromRole(newGR)
+	returnError = errors.Join(returnError, validateRules(globalRules, false, fldPath))
+	for _, rules := range newGR.NamespacedRules {
+		returnError = errors.Join(returnError, validateRules(rules, true, fldPath))
+	}
+	if returnError != nil {
+		return admission.ResponseBadRequest(returnError.Error()), nil
+	}
+
+	// Check that there are no escalations
+	hasEscalate := common.NewVerbChecker(request, newGR.Name, escalateVerb, a.sar, gvr)
 	clusterRules, err := a.grbResolver.GlobalRoleResolver.ClusterRulesFromRole(newGR)
 	if err != nil {
 		return nil, fmt.Errorf("unable to resolve rules for new global role: %w", err)
 	}
-
-	hasEscalate, err := a.isRulesAllowed(request, clusterRules, newGR.Name, a.grbResolver)
+	err = auth.ConfirmNoEscalation(request, clusterRules, "", a.grbResolver)
 	if err != nil {
-		return admission.ResponseFailedEscalation(err.Error()), nil
+		if hasEscalate("") {
+			return admission.ResponseAllowed(), nil
+		}
+		returnError = errors.Join(returnError, err)
 	}
-	if hasEscalate {
-		// if we have the escalate verb, no need to check global permissions
-		return admission.ResponseAllowed(), nil
+	err = auth.ConfirmNoEscalation(request, globalRules, "", a.resolver)
+	if err != nil {
+		if hasEscalate("") {
+			return admission.ResponseAllowed(), nil
+		}
+		returnError = errors.Join(returnError, err)
+	}
+	for namespace, rules := range newGR.NamespacedRules {
+		err = auth.ConfirmNoEscalation(request, rules, namespace, a.resolver)
+		if err != nil {
+			if hasEscalate(namespace) {
```

Comment:
Since GlobalRoles are cluster-scoped resources, you want to check if the user has "Escalate" at the cluster scope. Otherwise, users could take advantage of odd rbac configurations to give escalate in one namespace but not others, which we want to avoid.

Reply:
Could we add a test here to make the current implementation fail so that we don't introduce this mistake in the future?